src/web/pages/roles/RoleDetailsPage.tsx

```diff
@@ -143,53 +125,52 @@ const GeneralPermissions = ({permissions = [], links}) => {
   );
 };
 
-GeneralPermissions.propTypes = {
-  links: PropTypes.bool,
-  permissions: PropTypes.array,
-};
-
-const Page = ({
+const RoleDetailsPage = ({
   entity,
   generalPermissions = [],
-  links = true,
+  isLoading = true,
   permissions = [],
   onChanged,
   onDownloaded,
   onError,
-
-  ...props
-}) => {
+}: RoleDetailsPageProps) => {
   const [_] = useTranslation();
-
+  const navigate = useNavigate();
   return (
     <RoleComponent
       onCloneError={onError}
-      onCloned={goToDetails('role', props)}
-      onCreated={goToDetails('role', props)}
+      onCloned={goToDetails('role', navigate)}
+      onCreated={goToDetails('role', navigate)}
       onDeleteError={onError}
-      onDeleted={goToList('roles', props)}
+      onDeleted={goToList('roles', navigate)}
       onDownloadError={onError}
       onDownloaded={onDownloaded}
       onSaved={onChanged}
     >
-      {({clone, create, delete: delete_func, download, edit, save}) => (
-        <EntityPage
-          {...props}
+      {({clone, create, delete: deleteFunc, download, edit}) => (
+        <EntityPage<Role>
           entity={entity}
+          entityType="role"
+          isLoading={isLoading}
           sectionIcon={<RoleIcon size="large" />}
           title={_('Role')}
-          toolBarIcons={ToolBarIcons}
-          onRoleCloneClick={clone}
-          onRoleCreateClick={create}
-          onRoleDeleteClick={delete_func}
-          onRoleDownloadClick={download}
-          onRoleEditClick={edit}
-          onRoleSaveClick={save}
+          toolBarIcons={
+            <RoleDetailsPageToolBarIcons
+              entity={entity}
+              onRoleCloneClick={clone}
+              onRoleCreateClick={create}
+              onRoleDeleteClick={deleteFunc}
+              onRoleDownloadClick={download}
+              onRoleEditClick={edit}
+            />
+          }
         >
           {() => {
             return (
-              <React.Fragment>
-                <PageTitle title={_('Role: {{name}}', {name: entity.name})} />
+              <>
+                <PageTitle
+                  title={_('Role: {{name}}', {name: entity.name || ''})}
```

Comment:
should also always be defined here
```suggestion
                  title={_('Role: {{name}}', {name: entity.name as string})}
```